### PR TITLE
feat(core): shellbar improvements

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

closes https://github.com/SAP/fundamental-ngx/issues/7966

## Description
- Applied new styling of the shellbar;
- added ability to indirectly use platform search inside core shellbar;
- added additional styling for the search field inside shellbar;
- added ability to automatically detect fitting size of the shellbar if none was provided;
- added ability to move the search field inside shellbar between root of the shellbar and shellbar actions;
- added missing translations for shellbar;
- changed some content children to use lightweight tokens instead of component reference;

### Before:
image

### After:
image
image

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/shellbar/shellbar-actions-mobile/shellbar-actions-mobile.component.ts [hunk 1]
```diff
@@ -23,9 +25,17 @@ export class ShellbarActionsMobileComponent implements AfterContentChecked {
     @Input()
     collapsedItemMenuLabel: string;
 
+    /** @hidden */
+    @Input()
+    searchExists = false;
```
Comment (N1XUS) [inline libs/core/src/lib/shellbar/shellbar-actions-mobile/shellbar-actions-mobile.component.ts:30]: This component is internal one, so I think it's fine to not document some stuff here

Comment (platon-rov) [reply libs/core/src/lib/shellbar/shellbar-actions-mobile/shellbar-actions-mobile.component.ts:30]: Maybe we can mention this in comments? After hidden

File: libs/core/src/lib/shellbar/shellbar-actions/shellbar-actions.component.html [hunk 2]
```diff
@@ -2,17 +2,28 @@
     <ng-content select="fd-combobox"></ng-content>
 </div>
 
+<!-- <ng-container cdkPortalOutlet></ng-container> -->
```
Comment (platon-rov) [inline libs/core/src/lib/shellbar/shellbar-actions/shellbar-actions.component.html:5]: Please remove

File: libs/core/src/lib/shellbar/shellbar-actions/shellbar-actions.component.html [hunk 3]
```diff
@@ -2,17 +2,28 @@
...
+
+<!-- If portal outlet placed outside div, it will be added to the end of the root container. So this div actually keeps outlet in desired place. -->
+<div class="fd-shellbar__action" cdkPortalOutlet>
+    <!-- <ng-container cdkPortalOutlet></ng-container> -->
```
Comment (platon-rov) [inline libs/core/src/lib/shellbar/shellbar-actions/shellbar-actions.component.html:9]: And this

File: libs/core/src/lib/shellbar/shellbar.component.ts [hunk 4]
```diff
@@ -39,19 +74,168 @@ export class ShellbarComponent implements AfterContentInit {
...
+    /** @hidden */
+    @ContentChildren(FD_BUTTON_COMPONENT, { read: ElementRef })
+    buttons: QueryList<ElementRef>;
+
+    /** @hidden */
+    @ContentChild(ShellbarActionsComponent)
+    private _actions: ShellbarActionsComponent;
+
+    /** @hidden */
+    searchPortal: DomPortal;
```
Comment (platon-rov) [inline libs/core/src/lib/shellbar/shellbar.component.ts:89]: `_`  prefix?

**Review by platon-rov — CHANGES_REQUESTED** [review]
Looks cool!

Changes should be marked as breaking in the commit message.

**Comment by N1XUS** [comment]
@platon-rov thank you! Your comments are being addressed, please take a look.

**Review by platon-rov — APPROVED** [review]

**Pull request merged**